pkg/jaegerproxy/proxy.go

```diff
@@ -0,0 +1,161 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package jaegerproxy
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type ProxyConfig struct {
+	TLS                bool          `yaml:"promscale-tls,omitempty"`
+	CaFile             string        `yaml:"promscale-cafile,omitempty"`
+	ServerAddr         string        `yaml:"promscale-server,omitempty"`
+	ServerHostOverride string        `yaml:"promscale-server-host-override,omitempty"`
+	Timeout            time.Duration `yaml:"timeout,omitempty"`
```

Comment:
It seems that `Timeout` is never used. I guess we want to pass it down when establishing gRPC connection.

Reply:
yeah good point, I'll look into it